random-menu-generator.rb

```diff
@@ -0,0 +1,784 @@
+$how_many_ideas_desired = 0 # and I don't want any, either!
+$user_wants_too_many = false # zero just doesn't know how to be too many.
+
+#-------------------------- END WEIRD GLOBAL THINGS ----------------------------
+#--------------------- BEGIN OUTPUT TEXT TO USER THINGS ------------------------
+
+# outputs provided string of text to terminal & possibly to speakers as well via
+# %x interaction with outside terminal environment. !W discovered the hard way
+# that say does not take some punctuation well: & ' " \ (
+def speak(string_to_speak)
+  puts "#{string_to_speak}"
+
+  if $voice
+    good_voices = ["Alex", "Bad", "Good", "Vicki", "Zarvox"]
+    random_voice = good_voices[(rand * good_voices.length)]
+    %x{say -v #{random_voice} #{string_to_speak}}
+  end
+end
+
+
+
+
+# checks whether the voice should be toggle from true to false or vice versa by
+# looking for specific keywords in a provided string.
+def check_reset_voice(user_input)
+  voice_triggers = %w{speak word spoke say speech talk tell said}
+
+  voice_triggers.each do |voice_trigger|
+    if user_input.downcase.include? voice_trigger
+    # input is downcased in case the original get input call expected a number
+      if $voice == true
+        $voice = false
+      else
+        $voice = true
+      end
+    end
+  end
+end
+
+#---------------------- END OUTPUT TEXT TO USER THINGS -------------------------
+#----------------------------- BEGIN HELP THINGS -------------------------------
+
+# would help the user if only it knew how!
+# def help_user(user_input)
+#   # !W do something to help user here @_@
+#   speak("I'm sorry. I don't know how to help you yet.")
+#   speak("If you have any questions, please pass them along to Jeri.")
+#   speak("She can teach me how to be more helpful!")
+# end
+#
+#
+#
+#
+# # checks whether help_user should be called by looking for specific keywords in
+# # a provided string.
+# def check_help_request(user_input)
+#   help = %{help what how when why where who}
+#   help_triggers.each do |help_trigger|
+#     if user_input.include? help_trigger
+#       return help_user(user_input)
+#     end
+#   end
+# end
+#
+#------------------------------ END HELP THINGS --------------------------------
+#----------------- BEGIN VERIFICATION / CONFIRMATION THINGS --------------------
+
+# gets input from a user, passes the input to check_reset_voice, and then
+# returns the user input string back to whatever requested it. because this is
+# the only way user input is harvested, any time the user says something it is
+# possible to toggle the voice on/off or trigger the currently not-so-helpful
+# help_user function (!W). has an option to trigger .downcase! on the user input
+# string before returning it.
+def get_user_input(downcase)
+  user_input = $stdin.gets.chomp
+
+  if downcase
+    user_input.downcase!
+  end
+
+  check_reset_voice(user_input)
+  # check_help_request(user_input) #!W !T
+
+  return user_input
+end
```

Comment:
I'm noticing a number of conditionals in other functions that seem to serve the purpose of normalizing your user inputs. It might be nice to pull all your normalizing conditionals into a couple of different methods whose only purpose is to take an input and turn it into something that's easier to work with.